### PR TITLE
Replace object with TypeInfo for GetProperties()

### PR DESCRIPTION
**Pull request by jbagga**
Addresses #219 

cc @rynowak

### CODE REVIEW

File: test/Microsoft.Extensions.Internal.Test/PropertyHelperTest.cs [hunk 1]
```diff
@@ -111,8 +111,8 @@ public void PropertyHelper_ReturnsCachedPropertyHelper()
             var anonymous = new { foo = "bar" };
 
             // Act
-            var helpers1 = PropertyHelper.GetProperties(anonymous);
-            var helpers2 = PropertyHelper.GetProperties(anonymous);
```
Comment (rynowak) [inline test/Microsoft.Extensions.Internal.Test/PropertyHelperTest.cs]: At least one of these tests should test the new TypeInfo version

File: shared/Microsoft.Extensions.PropertyHelper.Sources/PropertyHelper.cs [hunk 2]
```diff
@@ -121,20 +121,20 @@ public void SetValue(object instance, object value)
...
-        /// <param name="instance">the instance to extract property accessors for.</param>
-        /// <returns>a cached array of all public property getters from the underlying type of target instance.
+        /// <param name="typeInfo">The type info to extract property accessors for.</param>
+        /// <returns>A cached array of all public property getters from the underlying type of target instance.
         /// </returns>
-        public static PropertyHelper[] GetProperties(object instance)
+        public static PropertyHelper[] GetProperties(TypeInfo typeInfo)
         {
-            return GetProperties(instance.GetType());
+            return GetProperties(typeInfo);
```
Comment (pranavkm) [inline shared/Microsoft.Extensions.PropertyHelper.Sources/PropertyHelper.cs]: Isn't this calling itself? You probably need `GetProperties(typeInfo.AsType())`

**Comment by jbagga** [comment]
@rynowak Should I also change `GetVisibleProperties(object instance)`?

**Comment by jbagga** [comment]
@rynowak @pranavkm Updated

File: shared/Microsoft.Extensions.PropertyHelper.Sources/PropertyHelper.cs [hunk 3]
```diff
@@ -121,20 +121,20 @@ public void SetValue(object instance, object value)
         /// Creates and caches fast property helpers that expose getters for every public get property on the
         /// underlying type.
         /// </summary>
-        /// <param name="instance">the instance to extract property accessors for.</param>
-        /// <returns>a cached array of all public property getters from the underlying type of target instance.
+        /// <param name="typeInfo">The type info to extract property accessors for.</param>
+        /// <returns>A cached array of all public property getters from the underlying type of target instance.
```
Comment (pranavkm) [inline shared/Microsoft.Extensions.PropertyHelper.Sources/PropertyHelper.cs]: > of all public properties of the specified type. 

(no instance anymore). That said, I'm ok if you removed the docs entirely too. It's a shared src type and doesn't need elaborate doc comments

File: shared/Microsoft.Extensions.PropertyHelper.Sources/PropertyHelper.cs [hunk 4]
```diff
@@ -151,13 +151,13 @@ public static PropertyHelper[] GetProperties(Type type)
         /// hidden by definitions using the <c>new</c> keyword.
         /// </para>
         /// </summary>
-        /// <param name="instance">The instance to extract property accessors for.</param>
+        /// <param name="typeInfo">The type info to extract property accessors for.</param>
         /// <returns>
         /// A cached array of all public property getters from the instance's type.
```
Comment (pranavkm) [inline shared/Microsoft.Extensions.PropertyHelper.Sources/PropertyHelper.cs]: No instance

**Pull request merged**